pkg/prometheus/server/statefulset.go

```diff
@@ -332,8 +332,13 @@ func makeStatefulSetSpec(
 		return nil, fmt.Errorf("failed to merge containers spec: %w", err)
 	}
 
+	serviceName := governingServiceName
+	if cpf.ServiceName != nil {
+		serviceName = *cpf.ServiceName
```

Comment:
Good catch! We missed in #6687 that the service name needs to be propagated to the statefulset spec :fearful: 

cc @mviswanathsai

Reply:
Whew! good catch indeed.